src/main/java/reposense/model/FileTypeManager.java

```diff
@@ -0,0 +1,103 @@
+package reposense.model;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.stream.Collectors;
+
+/**
+ * {@code FileTypeManager} is responsible for holding a list of whitelisted formats and user-specified custom
+ * groupings.
+ */
+public class FileTypeManager {
+    private static final String DEFAULT_GROUP = "other";
+
+    private List<FileType> formats;
+    private List<FileType> groups;
+
+    public FileTypeManager(List<FileType> formats) {
+        this.formats = formats;
+        groups = new ArrayList<>();
+    }
+
+    public String getFileType(String fileName) {
```

Comment:
I think name this `getFileTypeLabel` will be less confusing because you have a model name `FileType` plus, you have another function named `getFileTypeLabels`